### ISSUE TITLE
Videos (links) tab

### ISSUE THREAD

**Issue by sean8300** [sean8300/listerassister_issues#598]
Create a video tab so that dropbox links and URLs will be on the dashboard. If possible have "Video Ready" email messages that allow templated messages to be sent to agents (versus the editors copying and pasting messages into the messages tab). Mock-up sent via Slack 
 Submitted By:  Jessica Owen

**Comment by Juvie92** [comment]
![Screen Shot 2020-07-27 at 8 18 04 AM](https://user-images.githubusercontent.com/64864069/88729937-d2c05980-d0e9-11ea-9c58-3e04f004e114.png)

**Comment by Juvie92** [comment]
Will also need to create video tab on client side. 

Would need to display the various links that apply to that property/order.

**Comment by sean8300** [comment]
The screen shot is from the customers side, but you're correct that we need customer side for visibility and admin side to actually enter/manage.  Fields should probably be reactive on the customers side, meaning if we haven't entered a value for something it won't show up at all (would look kinda goofy with a ton of empty boxes on a photos only order).

We may want that tab to also be aware of whether or not *any* items are listing and say something like "No video services available for this property" or something to that effect, but lets come back to this once the core has been built

**Comment by Juvie92** [comment]
Client and admin side 'video tab' created. May need a few tweaks to get the look right. On client side only fields with a 'link' will show up. On admin side, all fields will show. Figured it would be an easy way to notice what and if anything is missing.

Just need email templates from Jessica to finish things up.

May need to keep a close eye on url and embed code lengths. In the database, urls are set to varchar100 and embed codes are set to varchar200. Works fine with examples I have to work with but not sure if that's always the case.

**Comment by Juvie92** [comment]
Waiting on email templates for Matterports.

**Comment by Juvie92** [comment]
New database table for video_links:

CREATE TABLE `lister_assister`.`video_links` ( `id` INT(10) NOT NULL AUTO_INCREMENT , `listing_id` INT(10) UNSIGNED NOT NULL , `drone_link` VARCHAR(100) NULL , `walkthrough_link` VARCHAR(100) NULL , `enhanced_walkthrough_link` VARCHAR(100) NULL , `matterport_branded_url` VARCHAR(100) NULL , `matterport_branded_embed` VARCHAR(200) NULL , `matterport_unbranded_url` VARCHAR(100) NULL , `matterport_unbranded_embed` VARCHAR(200) NULL , `matterport_expiration` VARCHAR(50) NULL , PRIMARY KEY (`id`)) ENGINE = InnoDB;

**Comment by Juvie92** [comment]
listing_id foreign key:

ALTER TABLE `video_links` ADD CONSTRAINT `fk_listing_id` FOREIGN KEY (`listing_id`) REFERENCES `listing`(`id`) ON DELETE RESTRICT ON UPDATE CASCADE;

**Comment by Juvie92** [comment]
I'll be adding new email templates to live so just need to update the codes for each one.

UPDATE email_template
SET code = 'drone_video_ready'
WHERE name = 'Drone Video Ready';

UPDATE email_template
SET code = 'wt_video_ready'
WHERE name = 'Walkthrough Video Ready';

UPDATE email_template
SET code = 'enh_wt_video_ready'
WHERE name = 'Enhanced WT Video Ready';

UPDATE email_template
SET code = 'matterport_ready'
WHERE name = 'Matterport Ready';

UPDATE email_template
SET code = 'matterport_w/mls_ready'
WHERE name = 'Matterport w/MLS Ready';

**Comment by Juvie92** [comment]
Will need a method of updating matterport expiration date based on options given to agents if they decide to extend their time.

![image](https://user-images.githubusercontent.com/64864069/89343145-b0ce5600-d658-11ea-9fe5-55a0fff5ceee.png)

and a manual method(datepicker) in case date isn't updated on the proper day.


*creating new ticket for this

**Comment by sean8300** [comment]
Sorry, but I'm going to completely blow this thing up.

Lets turn this into a table view on the dashboard.  

Change your video table to
id autoinc | listing_id listing.id | type string | link string | expires_ts int10 | is_deleted bool NULL
https://docs.google.com/spreadsheets/d/12eIlh0l6gs5nw-b1nlqUhb_xJBOnen2qLrdke5nCTVQ/edit?usp=sharing

On the admin dashboard change the video tab to a table, similar to below
![videoTable](https://user-images.githubusercontent.com/19527808/89676546-b3bb8780-d8a0-11ea-8f7e-501af271fc72.PNG)

View: just follows the link
Copy URL: copy the URL to clipboard (this can be finicky)
Enable/Disable: Simply switches the is_deleted trigger on the DB, anything is_deleted=true does not show up on customers side

The Add Video dialog will simply ask for the information listed in that type.  We'll have to get the full list of types but think itd just be drone, walkthrough, enhanced walkthrough, and matterport

I don't know what the embed code is, good chance its just the same HTML over and over again with just a unique number being the only difference from one Matterport to the next, which means we really only need to be asking for that unique ID number (and if its the same number for all Matterport link types then we only need to ask ocne and the system can generate the HTML for each item).

Probably put "send email ready" buttons in the table under the Actions heading for each item

For Matterport probably just have a button that says "copy embed code" / "copy branded embed code" buttons

On the customers side lets mimick the Inventory table to display each item
![customerSide](https://user-images.githubusercontent.com/19527808/89677072-963aed80-d8a1-11ea-94a6-56928b021e8e.PNG)

If possible maybe have a preview thumbnail where the LA icon is in the picture?  Dunno, don't waste too much time on this part, could get away with just having the Type, Link, Copy URL/Code actions.....prolly want to show the expiration date here as well

**Comment by Juvie92** [comment]
Will need to alter the 'video_links table a bit.

ALTER TABLE `video_links`
DROP `matterport_unbranded_url`
DROP `matterport_unbranded_embed`
DROP `matterport_branded_embed`

ALTER TABLE `video_links` 
CHANGE `matterport_branded_url` `matterport_url_slug` VARCHAR(100) CHARACTER SET latin1 COLLATE latin1_swedish_ci NULL DEFAULT NULL;

**Comment by sean8300** [comment]
I messed up and already merged with staging, going to go ahead and push to live to avoid having to mess around with undoing merges, but this is not working.  the matterport_url_slug saves, but nothing else.  
DB structure I've been given
listing_id
drone_link
walkthrough_link
enhanced_walkthrough_link
matterport_url_slug
matterport_expiration
